docs/synapse/userguides/storm_adv_control.rstorm

```diff
@@ -79,29 +110,49 @@ A few helpful tips when writing and debugging advanced Storm:
 
 **Be aware of your pipeline.**
 
-That is, understand what is in your current working set at any point in your query. A significant part of Storm troubleshooting comes down to figuring out that the current working set is not what you think it is.
+That is, understand what is in your **current working set** at any point in your query. A significant part of Storm
+troubleshooting comes down to figuring out that the current working set is not what you think it is.
 
-**Operations may execute multiple times.**
+**Be aware of your variables.**
+
+Storm supports both runtime-safe ("runtsafe") and non-runtime-safe ("non-runtsafe") variables. Non-runtsafe
+variables have values that may **change** based on the current node in the Storm pipeline. Another significant
```

Comment:
This is kinda tricky because a variable will always be runtsafe if it has a runtsafe value assigned to it at some point, but it can still change based on the current node in the pipeline. For example, in this query:
`inet:fqdn=vertex.link $foo=:domain`
`$foo` is not runtsafe, but in this query:
`$foo=bar inet:fqdn=vertex.link $foo=:domain`
`$foo` will be runtsafe since it was set to the runtsafe value `bar`.
`inet:fqdn=vertex.link $foo=:domain $foo=bar`
would also make `$foo` runtsafe

Reply:
That could be getting into "weird stuff with runtsafety" though and may be more complicated than necessary for this

Reply:
Reworded in an attempt to simplify - better?